src/appleseed.shaders/include/appleseed/color/as_color_transforms.h

```diff
@@ -0,0 +1,397 @@
+        RGB_to_XYZ[2] = vector(RGB_TO_XYZ_DCIP3_R2);
+    }
+    else if (color_space == "Rec.2020")
+    {
+        source_illuminant = "D65";
+
+        RGB_to_XYZ[0] = vector(RGB_TO_XYZ_REC2020_R0);
+        RGB_to_XYZ[1] = vector(RGB_TO_XYZ_REC2020_R1);
+        RGB_to_XYZ[2] = vector(RGB_TO_XYZ_REC2020_R2);
+    }
+    else if (color_space == "Rec.709")
+    {
+        source_illuminant = "D65";
+
+        RGB_to_XYZ[0] = vector(RGB_TO_XYZ_REC709_R0);
+        RGB_to_XYZ[1] = vector(RGB_TO_XYZ_REC709_R1);
+        RGB_to_XYZ[2] = vector(RGB_TO_XYZ_REC709_R2);
+    }
+    else
+    {
+#ifdef DEBUG
+        string shadername = "";
+        getattribute("shader:shadername", shadername);
+
+        warning("[WARNING]:Unsupported/unknown color space %s in %s, %s:%i\n",
+                color_space, shadername, __FILE__, __NAME__);
+#endif
+        exit(); // no color space nor illuminant, no point in continuing
+    }
+}
+
+void get_XYZ_to_RGB_matrix(
+    string color_space,
+    output vector XYZ_to_RGB[3],
+    output string source_illuminant)
+{
+    if (color_space == "ACES")
+    {
+        source_illuminant = "D60";
+
+        XYZ_to_RGB[0] = vector(XYZ_TO_RGB_ACES_R0);
+        XYZ_to_RGB[1] = vector(XYZ_TO_RGB_ACES_R1);
+        XYZ_to_RGB[2] = vector(XYZ_TO_RGB_ACES_R2);
+    }
+    else if (color_space == "ACEScg")
+    {
+        source_illuminant = "D60";
+
+        XYZ_to_RGB[0] = vector(XYZ_TO_RGB_ACESCG_R0);
+        XYZ_to_RGB[1] = vector(XYZ_TO_RGB_ACESCG_R1);
+        XYZ_to_RGB[2] = vector(XYZ_TO_RGB_ACESCG_R2);
+    }
+    else if (color_space == "AdobeRGB")
+    {
+        source_illuminant = "D65";
+
+        XYZ_to_RGB[0] = vector(XYZ_TO_RGB_ADOBERGB_R0);
+        XYZ_to_RGB[1] = vector(XYZ_TO_RGB_ADOBERGB_R1);
+        XYZ_to_RGB[2] = vector(XYZ_TO_RGB_ADOBERGB_R2);
+    }
+    else if (color_space == "DCIP3")
+    {
+        source_illuminant = "DCIP3";
+
+        XYZ_to_RGB[0] = vector(XYZ_TO_RGB_DCIP3_R0);
+        XYZ_to_RGB[1] = vector(XYZ_TO_RGB_DCIP3_R1);
+        XYZ_to_RGB[2] = vector(XYZ_TO_RGB_DCIP3_R2);
+    }
+    else if (color_space == "Rec.2020")
+    {
+        source_illuminant = "D65";
+
+        XYZ_to_RGB[0] = vector(XYZ_TO_RGB_REC2020_R0);
+        XYZ_to_RGB[1] = vector(XYZ_TO_RGB_REC2020_R1);
+        XYZ_to_RGB[2] = vector(XYZ_TO_RGB_REC2020_R2);
+    }
+    else if (color_space == "Rec.709")
+    {
+        source_illuminant = "D65";
+
+        XYZ_to_RGB[0] = vector(XYZ_TO_RGB_REC709_R0);
+        XYZ_to_RGB[1] = vector(XYZ_TO_RGB_REC709_R1);
+        XYZ_to_RGB[2] = vector(XYZ_TO_RGB_REC709_R2);
+    }
+    else
+    {
+#ifdef DEBUG
+        string shadername = "";
+        getattribute("shader:shadername", shadername);
+
+        warning("[WARNING]:Unsupported/unknown color space %s in %s, %s:%i\n",
+                color_space, shadername, __FILE__, __NAME__);
+#endif
+        exit(); // no color space nor illuminant, no point in continuing
```

Comment:
@dictoon it was intended since without a valid space or illuminant we cannot convert to XYZ, but now that i think about it, perhaps the best is making the noise active, not exclusively on debug builds, and set the matrix as identity.